pandas/io/parsers/base_parser.py

```diff
@@ -762,8 +763,15 @@ def _infer_types(
             pa = import_optional_dependency("pyarrow")
             if isinstance(result, np.ndarray):
                 result = ArrowExtensionArray(pa.array(result, from_pandas=True))
+            elif isinstance(result, BaseMaskedArray):
+                if result._mask.all():
+                    result = result.to_numpy(na_value=None)
```

Comment:
Why do we need this conversion first for the `mask.all()` case?

Reply:
We want a null pyarrow array here, not sure how we can get there otherwise

Reply:
done

Reply:
Or maybe it's not even worth doing `to_numpy`, just skipping to `ArrowExtensionArray(pa.array([None]* len(result)))`